src/core/agent/modules/signify/signifyApi.ts

```diff
@@ -105,6 +105,39 @@ export class SignifyApi {
     }
   }
 
+  async createDelegationIdentifier(delegatorPrefix: string): Promise<string> {
+    const signifyName = utils.uuid();
+    const icpResult2 = await this.signifyClient
+      .identifiers()
+      .create(signifyName, { delpre: delegatorPrefix });
+    const operation = await icpResult2.op();
+    // no need to wait for operation to complete because it needs to be interacted from delegator
+    const delegatePrefix = operation.name.split(".")[1];
+    return delegatePrefix;
```

Comment:
Sorry, my ACs weren't so clear - so this is perfect for `signifyApi.ts` but I also need the interfaces in `identifierService.ts` that makes it available to the UI.

In addition, the service should create pending metadata records, and mark the record as complete once the delegation has been completed at the other side.

For now, the metadata record only needs to be created for the wallet that initiates and creates the identifier. For the wallet that holds the identifier, we can cover this later when we understand it better.

Reply:
Could you help me to check it again?